common/error.hpp

```diff
@@ -53,8 +53,8 @@ class Error {
 		return this->message == other.message && this->code == other.code;
 	}
 
-	operator bool() const {
-		return static_cast<bool>(this->code);
+	bool operator!=(const Error &other) const {
+		return this->message != other.message || this->code != other.code;
```

Comment:
How does this work in comparisons with `error::NoError` when the `code` is `error_condition` and thus has both a category and an integer code?

Reply:
Yes, we return error::NoError in all the places.
I've cleaned up the commits.